### DISCUSSION TITLE
Name backups with Mealie release version?

### DISCUSSION THREAD

**Discussion by mgrecar** [mealie-recipes/mealie#4659] [Ideas]
I was doing my first upgrade today from v2.2.0 to v2.3.0 when I noticed that Mealie backup filename only includes the timestamp of the backup.  Given the possibility of breaking database changes between versions, as well as just at a glance clarity of under what circumstance a backup was taken, would putting the version of Mealie in the backup file name be a good idea?

I manually renamed mine like `mealie_v2.2.0_2024.12.02.15.06.32.zip`, but system doing something like that automatically would be useful.  I think visual confirmation of a version compatible backup existing would help.

**Comment by michael-genson** [comment]
Yup I think this makes sense. It won't be too useful for nightly but it can be useful for everyone else.
I guess nightly could include the git hash.

**Comment by Kuchenpirat** [comment]
I think this makes sense, but also this is normally not needed since you are able to import backups from older versions of mealie. 

The database will execute all migrations between the import and the currently installed version automatically as long as the installed version is on the same db version or newer.

**Reply by mgrecar** [reply]
> I think this makes sense, but also this is normally not needed since you are able to import backups from older versions of mealie.
> 
> The database will execute all migrations between the import and the currently installed version automatically as long as the installed version is on the same db version or newer.

Oh, I was not aware of that.  That's pretty useful, and alleviates my chief worry about the situation.  I still think it might be useful as a visual data point about taking a proper backup before an update, but it's way less of a concern for me now.
